### PR TITLE
Chocrates/release/version 2.0

### PR DESCRIPTION
**Pull request by Chocrates**
Changes to get ldap working and updates teams on team.create

### CODE REVIEW

File: githubapp/ldap.py [hunk 1]
```diff
@@ -47,7 +50,11 @@ def get_group_members(self, group_name):
             if entry['type'] == 'searchResEntry':
                 for member in entry['attributes'][self.LDAP_GROUP_MEMBER_ATTRIBUTE]:
                     try:
-                        member_list.append(self.get_attr_by_dn(member))
+                        if 'uid' in member:
+                            member_dn = member
+                        else:
+                            member_dn = f'uid={member},{self.LDAP_USER_BASE_DN}'
+                        member_list.append(self.get_attr_by_dn(member_dn))
```
Comment (primetheus) [inline githubapp/ldap.py:57]: this breaks Active Directory lookups... probably need to break out a new method that handles the distinguishing of LDAP entry types 🤔

File: githubapp/ldap.py [hunk 2]
```diff
@@ -14,7 +14,10 @@ def __init__(self):
         self.LDAP_GROUP_BASE_DN = os.environ['LDAP_GROUP_BASE_DN']
         self.LDAP_GROUP_FILTER = os.environ['LDAP_GROUP_FILTER']
         self.LDAP_GROUP_MEMBER_ATTRIBUTE = os.environ['LDAP_GROUP_MEMBER_ATTRIBUTE']
-        self.LDAP_BIND_USER = os.environ['LDAP_BIND_USER']
+        if 'bind_user' in os.environ:
```
Comment (primetheus) [inline githubapp/ldap.py:17]:
```suggestion
        if 'LDAP_BIND_USER' in os.environ:
```

that works... do we want to standardize on one or the other? I chose `USER` arbitrarily

File: githubapp/ldap.py [hunk 3]
```diff
@@ -47,7 +50,11 @@ def get_group_members(self, group_name):
             if entry['type'] == 'searchResEntry':
                 for member in entry['attributes'][self.LDAP_GROUP_MEMBER_ATTRIBUTE]:
                     try:
-                        member_list.append(self.get_attr_by_dn(member))
+                        if 'uid' in member:
```
Comment (primetheus) [inline githubapp/ldap.py:53]:
```suggestion
                        if any(attr in member.casefold() for attr in ['uid', 'cn'])::
```

I haven't tested this with OpenLDAP, but it works with AD now

File: githubapp/ldap.py (hunk 2 above)
Comment (primetheus) [reply githubapp/ldap.py:17]: actually, with `user@domain.com` it wouldn't make sense to call it a `DN`... maybe we just leave it with both

File: githubapp/ldap.py [hunk 4]
```diff
@@ -14,7 +14,10 @@ def __init__(self):
         self.LDAP_GROUP_BASE_DN = os.environ['LDAP_GROUP_BASE_DN']
         self.LDAP_GROUP_FILTER = os.environ['LDAP_GROUP_FILTER']
         self.LDAP_GROUP_MEMBER_ATTRIBUTE = os.environ['LDAP_GROUP_MEMBER_ATTRIBUTE']
-        self.LDAP_BIND_USER = os.environ['LDAP_BIND_USER']
+        if 'bind_user' in os.environ:
+            self.LDAP_BIND_USER = os.environ['LDAP_BIND_USER']
+        else: 
+            self.LDAP_BIND_USER = os.environ['LDAP_BIND_DN']
```
Comment (primetheus) [inline githubapp/ldap.py:20]:
```suggestion
        if 'LDAP_BIND_USER' in os.environ:
            self.LDAP_BIND_USER = os.environ['LDAP_BIND_USER']
        elif 'LDAP_BIND_DN' in os.environ:
            self.LDAP_BIND_USER = os.environ['LDAP_BIND_DN']
        else:
            raise Exception('LDAP credentials have not been specified')
```

File: githubapp/ldap.py (hunk 2 above)
Comment (Chocrates) [reply githubapp/ldap.py]: Yeah I don't mind either way as long as add a comment about it

**Comment by Chocrates** [comment]
This works for me now in LDAP as well!

**Pull request merged**